src/collective/volto/formsupport/restapi/deserializer/blocks.py

```diff
@@ -25,6 +47,7 @@ def __call__(self, value):
         if value.get("send_message", ""):
             transform = SafeHTML()
             value["send_message"] = transform.scrub_html(value["send_message"])
+        value = update_global_forms(value)
```

Comment:
Let me better understand this..when you deserialize the block, you are storing its data twice: in the context and in registry, right?
So if the block is in a context, its data is duplicated? Or am i missing something?

I'm not a big fan of this approach, but i don't know if there is a better way to do it, or avoid setting it globally when we are working on a content item.

Or (maybe better): store form data always in registry, and in the content, only store the block_id reference. What do you think about it?

Reply:
Personally I think storing both the saved data and the schema on the root object is the way to go, regardless of if there is a local page context or not. Then it's one implementation so less complicated.
This would mean that
- its possible to move a form between pages and keep the data
- it's possible have two forms that submit to the same savedata. For example a slightly altered one with extra info for logged in users
- it's possible to have a control panel that lets you view all saved data globally and remove it. giving some kind of audibility to collected data.
   - but this does have implications for permissions. because then a private form on a private page is no longer private...